### DISCUSSION TITLE
Linux: curl version issue

### DISCUSSION THREAD

**Discussion by danb123** [Homebrew/discussions#4288] [Linux]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
Error: Please update your system curl or set HOMEBREW_CURL_PATH to a newer version.
Minimum required version: 7.41.0
Your curl version: 7.29.0
Your curl executable:
HOMEBREW_VERSION: 4.0.4-86-g7c15dce
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7c15dce285900c85476d445dcbc01418e61be37c
Last commit: 12 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 41165290bead61f90c288fedd4f8432296cd60a0
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /home/dbarrowman/.linuxbrew
HOMEBREW_REPOSITORY: /home/dbarrowman/.linuxbrew/Homebrew
HOMEBREW_CELLAR: /home/dbarrowman/.linuxbrew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 24
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 2.6.8 => /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: 24-core 64-bit zen
Clang: N/A
Git: 2.38.1 => /bin/git
Curl: 7.29.0 => /bin/curl
Kernel: Linux 3.10.0-693.2.2.el7.x86_64 x86_64 GNU/Linux
OS: CentOS Linux release 7.4.1708 (Core)  (Core)
Host glibc: 2.17
/usr/bin/gcc: 4.8.5
/usr/bin/ruby: 2.0.0
glibc: 2.35_1
gcc@11: N/A
gcc: N/A
xorg: N/A
```

### Output of `brew doctor`

```shell
Error: Please update your system curl or set HOMEBREW_CURL_PATH to a newer version.
Minimum required version: 7.41.0
Your curl version: 7.29.0
Your curl executable:
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/boost/python/detail/python22_fixed.h
  /usr/local/include/boost/regex.h
  /usr/local/include/boost/tr1/tr1/bcc32/array.h
  /usr/local/include/boost/tr1/tr1/bcc32/random.h
  /usr/local/include/boost/tr1/tr1/bcc32/regex.h
  /usr/local/include/boost/tr1/tr1/bcc32/tuple.h
  /usr/local/include/boost/tr1/tr1/bcc32/type_tra.h
  /usr/local/include/boost/tr1/tr1/bcc32/unordere.h

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libboost_atomic.a
  /usr/local/lib/libboost_chrono.a
  /usr/local/lib/libboost_context.a
  /usr/local/lib/libboost_coroutine.a
  /usr/local/lib/libboost_date_time.a
  /usr/local/lib/libboost_exception.a
  /usr/local/lib/libboost_filesystem.a
  /usr/local/lib/libboost_graph.a
  /usr/local/lib/libboost_iostreams.a
  /usr/local/lib/libboost_locale.a
  /usr/local/lib/libboost_log.a
  /usr/local/lib/libboost_log_setup.a
  /usr/local/lib/libboost_math_c99.a
  /usr/local/lib/libboost_math_c99f.a
  /usr/local/lib/libboost_math_c99l.a
  /usr/local/lib/libboost_math_tr1.a
  /usr/local/lib/libboost_math_tr1f.a
  /usr/local/lib/libboost_math_tr1l.a
  /usr/local/lib/libboost_prg_exec_monitor.a
  /usr/local/lib/libboost_program_options.a
  /usr/local/lib/libboost_random.a
  /usr/local/lib/libboost_regex.a
  /usr/local/lib/libboost_serialization.a
  /usr/local/lib/libboost_signals.a
  /usr/local/lib/libboost_system.a
  /usr/local/lib/libboost_test_exec_monitor.a
  /usr/local/lib/libboost_thread.a
  /usr/local/lib/libboost_timer.a
  /usr/local/lib/libboost_unit_test_framework.a
  /usr/local/lib/libboost_wave.a
  /usr/local/lib/libboost_wserialization.a

Warning: Your Homebrew's prefix is not /home/linuxbrew/.linuxbrew.

Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from Homebrew or its maintainers on social media.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

I guess I have a "chicken and the egg" problem. Unfortunately I'm on a system without root access. This is actually why I want to use Linux homebrew. The existing version of curl on the system is not current enough for brew. I did install a newer version of curl in my home directory and then pointed brew to that using HOMEBREW_CURL_PATH, but, that fails completely since the versions of curl and libcurl do not match. I don't think I can easily install or run my own version of libcurl?

Homebrew has the option to install curl as well, but, that install itself also needs an existing, new enough, version of curl, already on the system.

Thanks,
Dan

**Comment by gromgit** [comment]
CentOS 7.4 is almost 6 years old as of this writing, so the following may or may not work:
```sh
mkdir -p ~/tmp

# Download a modern statically-linked curl
curl -L -o ~/tmp/static-curl https://github.com/moparisthebest/static-curl/releases/download/v7.88.1/curl-amd64
chmod +x ~/tmp/static-curl

# Check that you can actually run this binary
~/tmp/static-curl --version

# See if Homebrew accepts it
HOMEBREW_CURL_PATH=${HOME}/tmp/static-curl brew config

# If so, install Homebrew curl with it...
HOMEBREW_CURL_PATH=${HOME}/tmp/static-curl brew install curl

# ...then see if Homebrew's happy to use its own curl
brew config
```

**Reply by danb123** [reply]
Thanks much. I think that got me a bit further but now there is an issue with gcc. Here are some results:

dbarrowman@stark02-vm-lnode02:~ $ ~/tmp/static-curl --version
curl 7.88.1 (x86_64-pc-linux-musl) libcurl/7.88.1 OpenSSL/1.1.1t zlib/1.2.12 libssh2/1.10.0 nghttp2/1.47.0
Release-Date: 2023-02-20
Protocols: dict file ftp ftps gopher gophers http https imap imaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS HSTS HTTP2 HTTPS-proxy IPv6 Largefile libz NTLM NTLM_WB SSL threadsafe TLS-SRP UnixSockets
dbarrowman@stark02-vm-lnode02:~ $ HOMEBREW_CURL_PATH=${HOME}/tmp/static-curl brew config
HOMEBREW_VERSION: 4.0.4-86-g7c15dce
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7c15dce285900c85476d445dcbc01418e61be37c
Last commit: 18 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: bfaf49746e3d82ba3734dcd18e13a7cd582f1df8
Core tap last commit: 3 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /home/dbarrowman/.linuxbrew
HOMEBREW_REPOSITORY: /home/dbarrowman/.linuxbrew/Homebrew
HOMEBREW_CELLAR: /home/dbarrowman/.linuxbrew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_CURL_PATH: /home/dbarrowman/tmp/static-curl
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 24
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 2.6.8 => /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: 24-core 64-bit zen
Clang: N/A
Git: 2.38.1 => /bin/git
Curl: 7.88.1 => /home/dbarrowman/tmp/static-curl
Kernel: Linux 3.10.0-693.2.2.el7.x86_64 x86_64 GNU/Linux
OS: CentOS Linux release 7.4.1708 (Core)  (Core)
Host glibc: 2.17
/usr/bin/gcc: 4.8.5
/usr/bin/ruby: 2.0.0
glibc: 2.35_1
gcc@11: N/A
gcc: N/A
xorg: N/A
dbarrowman@stark02-vm-lnode02:~ $ gcc --version
gcc (GCC) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

**Reply by gromgit** [reply]
Have you actually run the last two commands I mentioned?

**Reply by danb123** [reply]
Yes, sorry I didn't show that. Here are the results. The first command seemed to run fine:

dbarrowman@stark02-vm-lnode02:~ $ HOMEBREW_CURL_PATH=${HOME}/tmp/static-curl brew config
HOMEBREW_VERSION: 4.0.4-86-g7c15dce
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7c15dce285900c85476d445dcbc01418e61be37c
Last commit: 18 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 41165290bead61f90c288fedd4f8432296cd60a0
Core tap last commit: 7 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /home/dbarrowman/.linuxbrew
HOMEBREW_REPOSITORY: /home/dbarrowman/.linuxbrew/Homebrew
HOMEBREW_CELLAR: /home/dbarrowman/.linuxbrew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_CURL_PATH: /home/dbarrowman/tmp/static-curl
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 24
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 2.6.8 => /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: 24-core 64-bit zen
Clang: N/A
Git: 2.38.1 => /bin/git
Curl: 7.88.1 => /home/dbarrowman/tmp/static-curl
Kernel: Linux 3.10.0-693.2.2.el7.x86_64 x86_64 GNU/Linux
OS: CentOS Linux release 7.4.1708 (Core)  (Core)
Host glibc: 2.17
/usr/bin/gcc: 4.8.5
/usr/bin/ruby: 2.0.0
glibc: 2.35_1
gcc@11: N/A
gcc: N/A
xorg: N/A

Here are the tail results of the 2nd command showing the error. The run output was quite long so I couldn't include all of it.

==> Fetching curl
==> Downloading https://ghcr.io/v2/homebrew/core/curl/manifests/7.88.1
Already downloaded: /home/dbarrowman/.cache/Homebrew/downloads/df1deb709cbf6f6ab7075358916421ef97d1d09cb6dbcb1b9e6bd9c5f194c4ff--curl-7.88.1.bottle_manifest.json
==> Downloading https://ghcr.io/v2/homebrew/core/curl/blobs/sha256:f915f992603842a7f608881e78d696fb9fd288e7fcf0f9f9c2b76fb72c2508e4
Already downloaded: /home/dbarrowman/.cache/Homebrew/downloads/5a2f86ec429b816fbbe671808bc57c1a6a469efa1f19cff981c4b9a957dc74e6--curl--7.88.1.x86_64_linux.bottle.tar.gz
==> Installing dependencies for curl: xz, zlib, zstd, binutils, gcc, brotli, pkg-config, libunistring, libidn2, libnghttp2, ca-certificates, openssl@1.1, libssh2, libxcrypt, gpatch, ncurses, readline, util-linux, openldap, rtmpdump, m4, bison, libedit and krb5
==> Installing curl dependency: xz
==> ./configure --prefix=/home/dbarrowman/.linuxbrew/Cellar/xz/5.4.1
Last 15 lines from /home/dbarrowman/.cache/Homebrew/Logs/xz/01.configure:

Initializing Automake:
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether ln -s works... yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... no
configure: error: in `/tmp/xz-20230301-24720-1w4j76r/xz-5.4.1':
configure: error: C compiler cannot create executables
See `config.log' for more details

READ THIS: https://docs.brew.sh/Troubleshooting

/home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/utils/github/api.rb:317:in `raise_error': curl failed!   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current (GitHub::API::Error)
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (77) error setting certificate verify locations:  CAfile: /etc/ssl/certs/ca-certificates.crt CApath: none
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/utils/github/api.rb:241:in `open_rest'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/utils/github.rb:186:in `search'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/utils/github.rb:190:in `search_results_items'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/utils/github.rb:40:in `search_issues'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/utils/github.rb:70:in `issues_for_formula'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/exceptions.rb:498:in `fetch_issues'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/exceptions.rb:493:in `issues'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/exceptions.rb:549:in `dump'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:150:in `rescue in <main>'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:138:in `<main>'
/home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2538:in `block in system': Failed executing: ./configure --disable-debug --disable-dependency-tracking --disable-silent-rules --prefix=/home/dbarrowman/.linuxbrew/Cellar/xz/5.4.1 (BuildError)
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2474:in `open'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2474:in `system'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/xz.rb:30:in `install'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/build.rb:177:in `block (3 levels) in install'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/extend/kernel.rb:525:in `with_env'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/build.rb:139:in `block (2 levels) in install'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1325:in `block in brew'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2705:in `block (2 levels) in stage'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/extend/kernel.rb:525:in `with_env'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2704:in `block in stage'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:158:in `block (2 levels) in unpack'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/download_strategy.rb:116:in `chdir'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/download_strategy.rb:116:in `chdir'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/download_strategy.rb:103:in `stage'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:154:in `block in unpack'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:77:in `block in run'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:77:in `chdir'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/mktemp.rb:77:in `run'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:266:in `stage_resource'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:153:in `unpack'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:127:in `stage'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/forwardable.rb:230:in `stage'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:2684:in `stage'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1318:in `brew'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/build.rb:133:in `block in install'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/extend/kernel.rb:525:in `with_env'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/build.rb:128:in `install'
        from /home/dbarrowman/.linuxbrew/Homebrew/Library/Homebrew/build.rb:229:in `<main>'

**Reply by danb123** [reply]
I set the environment variable:
dbarrowman@stark02-vm-lnode02:~/.cache/Homebrew $ export HOMEBREW_CURL_PATH="${HOME}/tmp/static-curl"
dbarrowman@stark02-vm-lnode02:~/.cache/Homebrew $ printenv | grep HOMEBREW
HOMEBREW_REPOSITORY=/home/dbarrowman/.linuxbrew/Homebrew
HOMEBREW_CURL_PATH=/home/dbarrowman/tmp/static-curl
HOMEBREW_PREFIX=/home/dbarrowman/.linuxbrew
HOMEBREW_CELLAR=/home/dbarrowman/.linuxbrew/Cellar

but then tried installing something and it still didn't work:

dbarrowman@stark02-vm-lnode02:~ $ brew install tree
==> Fetching dependencies for tree: xz, zlib, zstd, binutils and gcc
==> Fetching xz
Warning: Building xz from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/dbarrowman/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/dbarrowman/.linuxbrew)
==> Downloading https://downloads.sourceforge.net/project/lzmautils/xz-5.4.1.tar.gz
Already downloaded: /home/dbarrowman/.cache/Homebrew/downloads/ca0c062e8ecb14160415874ad277c80d73d9a9d00aa926ff11a32e877f6651fe--xz-5.4.1.tar.gz
==> Fetching zlib
==> Downloading https://ghcr.io/v2/homebrew/core/zlib/manifests/1.2.13-1
curl: (77) error setting certificate verify locations:  CAfile: /etc/ssl/certs/ca-certificates.crt CApath: none

Error: tree: Failed to download resource "zlib_bottle_manifest"
Download failed: https://ghcr.io/v2/homebrew/core/zlib/manifests/1.2.13-1

**Reply by danb123** [reply]
Tried setting HOMEBREW_FORCE_BREWED_CURL as well:

dbarrowman@stark02-vm-lnode02:~ $ export HOMEBREW_FORCE_BREWED_CURL=1
dbarrowman@stark02-vm-lnode02:~ $ brew install tree
Running `brew update --auto-update`...
==> Fetching dependencies for curl: xz, zlib, zstd, binutils, gcc, brotli, pkg-config, libunistring, libidn2, libnghttp2, ca-certificates, openssl@1.1, libssh2, libxcrypt, gpatch, ncurses, readline, util-linux, openldap, rtmpdump, m4, bison, libedit and krb5
==> Fetching xz
Warning: Building xz from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/dbarrowman/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/dbarrowman/.linuxbrew)
==> Downloading https://downloads.sourceforge.net/project/lzmautils/xz-5.4.1.tar.gz
Already downloaded: /home/dbarrowman/.cache/Homebrew/downloads/ca0c062e8ecb14160415874ad277c80d73d9a9d00aa926ff11a32e877f6651fe--xz-5.4.1.tar.gz
==> Fetching zlib
==> Downloading https://ghcr.io/v2/homebrew/core/zlib/manifests/1.2.13-1
curl: (77) error setting certificate verify locations:  CAfile: /etc/ssl/certs/ca-certificates.crt CApath: none

Error: curl: Failed to download resource "zlib_bottle_manifest"
Download failed: https://ghcr.io/v2/homebrew/core/zlib/manifests/1.2.13-1
Error: 'curl' must be installed and in your PATH!
==> Fetching dependencies for tree: xz, zlib, zstd, binutils and gcc
==> Fetching xz
Warning: Building xz from source as the bottle needs:
- HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar (yours is /home/dbarrowman/.linuxbrew/Cellar)
- HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew (yours is /home/dbarrowman/.linuxbrew)
==> Downloading https://downloads.sourceforge.net/project/lzmautils/xz-5.4.1.tar.gz
Already downloaded: /home/dbarrowman/.cache/Homebrew/downloads/ca0c062e8ecb14160415874ad277c80d73d9a9d00aa926ff11a32e877f6651fe--xz-5.4.1.tar.gz
==> Fetching zlib
==> Downloading https://ghcr.io/v2/homebrew/core/zlib/manifests/1.2.13-1
curl: (77) error setting certificate verify locations:  CAfile: /etc/ssl/certs/ca-certificates.crt CApath: none

Error: tree: Failed to download resource "zlib_bottle_manifest"
Download failed: https://ghcr.io/v2/homebrew/core/zlib/manifests/1.2.13-1

**Reply by gromgit** [reply]
At this point, you'll have to ask yourself "how badly do I want Homebrew"? You'll be paying a very heavy price for not being able to install Homebrew in the officially sanctioned location.

Some formulae have relocatable bottles (e.g. `gcc`, I think, which is a good thing--you really don't want to have to build GCC from scratch). For those like `xz` which don't, you'll have to read through the build logs to see what went wrong, and figure out how to fix them. (On Linux, the build logs for `<formula>` are generally in `~/.cache/Homebrew/Logs/<formula>/`.)

One alternative is to decide exactly what software you need, then download the source tarballs and build them from scratch. You can then configure them with exactly as many features as you require. (Homebrew `curl` has a rather long list of dependencies because it's meant to cater to a large audience, so it enables quite a few things that may not be relevant to you.) You'd also have to use a package versioning tool like [graft](https://peters.gormand.com.au/Home/tools/graft/) to organize your software installs, so that long-term maintenance doesn't drive you up the wall.

Either way, you'll have to get comfortable with building from source, and fixing whatever issues crop up.

**Reply by danb123** [reply]
Yes I agree with what you are saying. This all resulted from me just looking for a way to install certain software and trying to find a way to do that. So it's more of a general problem, and yes, I don't want to have some weird, unstable config that can constantly have issues.

I have already installed some things from source so I have a general idea of how to do that. This has also been a good learning experience anyway. Something for the future that I can use for my home setup where I have more control and not this locked down work environment with an ancient Centos image and no root access. :)

But I really appreciate you taking the time and effort to help me and you have definitely helped me at least learn some new things.
